client/packages/invoices/src/InboundShipment/DetailView/SidePanel/TransportSection.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC, memo } from 'react';
+import {
+  Grid,
+  DetailPanelSection,
+  PanelLabel,
+  useTranslation,
+  useBufferState,
+  BufferedTextInput,
+} from '@openmsupply-client/common';
+import { useInbound } from '../../api';
+
+export const TransportSectionComponent: FC = () => {
+  const t = useTranslation('distribution');
+  const { transportReference, update } = useInbound.document.fields([
+    'transportReference',
+  ]);
+  const [referenceBuffer, setReferenceBuffer] = useBufferState(
+    transportReference ?? ''
+  );
+
+  // the text input is read-only, as the transport reference is only populated
+  // as part of the transfer creation process,
+  // it isn't possible currently for users to edit it
+
+  return (
+    <DetailPanelSection title={t('heading.transport-details')}>
+      <Grid container gap={0.5} key="transport-details">
+        <PanelLabel display="flex" alignItems="center">
+          {t('heading.reference')}
+        </PanelLabel>
+        <BufferedTextInput
+          disabled={true}
+          onChange={e => {
+            setReferenceBuffer(e.target.value);
+            update({ transportReference: e.target.value });
+          }}
```

Comment:
Could you please explain to me why this is needed/ why you do it like this? e.g. why is setReferenceBuffer called in onChange? and why call update()? isn't the e.target.value always the same as what you put into the text input?

Reply:
I believe the issue is with managing updates to the document, in this case the invoice. The problem is, if you are using only the internal state of the text component and there is an issue with the async update of the document then the value shown in the textfield won't reflect what is actually stored in the API - if the API update fails, then the API will have `value_a` while the component displays `value_b`
An alternative approach is to set the component to be read only when updating ( painful to use though when you are trying to have the component update async with a debounce callback ) or to have the value driven only by the external value ( which requires constant updating ), or to only update externally when the component loses focus ( increases the potential for the entered value to not be saved, or for dependent actions to not fire when expected, such as form validation ).